internal/api/identity.go

```diff
@@ -109,6 +112,34 @@ func (a *API) linkIdentityToUser(ctx context.Context, tx *storage.Connection, us
 	if _, terr := a.createNewIdentity(tx, targetUser, providerType, structs.Map(userData.Metadata)); terr != nil {
 		return nil, terr
 	}
+
+	if targetUser.GetEmail() == "" {
+		if terr := targetUser.UpdateUserEmail(tx); terr != nil {
+			if models.IsUniqueConstraintViolatedError(terr) {
```

Comment:
Also how would it be a unique violation error if the email is empty?

Reply:
@hf the `UpdateUserEmail` function can be named better, but it basically updates the auth.users.email column with the identity's email